xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp

```diff
@@ -1964,7 +1964,19 @@ std::string CDVDDemuxFFmpeg::GetStreamCodecName(int iStreamId)
 
     AVCodec* codec = avcodec_find_decoder(stream->codec);
     if (codec)
-      strName = codec->name;
+    {
+      switch (codec->id)
```

Comment:
I think we should instead just use `avcodec_get_name(codec->id)`. This would return `av1` for `AV_CODEC_ID_AV1`.

The full list of descriptors is here https://ffmpeg.org/doxygen/4.0/codec__desc_8c_source.html#l03135

I'm not sure why we don't do that already :confused:

Reply:
You are right, that would be a lot better. I'll change and test this tomorrow.

Reply:
I pushed an updated commit. Works like a charm. I tested it with a `h.264` and a `AV1` video, which were both displayed correctly.

Reply:
@ronie You probably want to also test some other formats. I checked the list in https://github.com/FFmpeg/FFmpeg/blob/n4.2.2/libavcodec/codec_desc.c and found matching flags for the most important code-names.

I think it wasn't a feature to display the actual FFmpeg decoder-name instead of the codec-name. For example: FFmpeg has two h264 decoders:
* https://github.com/FFmpeg/FFmpeg/blob/n4.2.2/libavcodec/h264dec.c#L1051
* https://github.com/FFmpeg/FFmpeg/blob/n4.2.2/libavcodec/libopenh264dec.c#L164

But there is no `libopenh264` flag in the skins (probably because this decoder isn't used by default).

I think `AV1` might be the actual first codec were there isn't a default decoder.

Reply:
thanx for the heads-up, i'll keep an eye on it.

skins don't (ofc.) includes flags for each and every codec, only the most common ones are present. i'm sure users will notify us if certain flags are suddenly missing. those can be added upon request.

Reply:
often users have pointed out of missing h265 mediaflag used for VP9/HEVC type content
but i don't understand mediaflag works with InputStreamAdaptive
since we're on the subject i report it here